compiler-plugin/src/main/java/arrow/plugin/MetaClassBuilderInterceptorExtension.kt

```diff
@@ -73,6 +73,13 @@ internal class MetaClassBuilder(
       override fun visitElement(element: PsiElement?) {
         if (element is KtCallExpression) {
           checkExpressionPurity(descriptor, element)
+        } else if (element is KtThrowExpression) {
+          messageCollector.report(
+            ERROR,
+            "Impure expression in function ${descriptor.name}: `${element.text}` returning `Nothing` due to `throw` expression only allowed in `suspend` functions.\n" +
```

Comment:
Why exceptions thrown in `suspend` functions are considered "safe"? Aren't there some scenarios on coroutines where you're enforced to hook error handling strategies to the job or it ends up crashing everything?

Reply:
Thoughts @raulraja? does it make sense to consider exceptions unsafe, no matter they're thrown in suspended or not suspended contexts? (so we enforce users to work more functionally and return proper error results types) 

Is that doable tho 🤔? libraries throw (let's say okhttp) and those might be causing compilation to fail if we consider exceptions within suspended contexts also as a compilation error?

Reply:
Exceptions are unsafe but also an effect. There are terminal cases where the user may want to throw and in those cases the only thing we can do is ask to be suspended so we guarantees the IO loop will capture those